index.bs

```diff
@@ -1598,286 +1600,161 @@ A URL for Blob and File reference</h2>
-user agents use a Content-Type header [[RFC7231]]
-that is equal to the value of the {{Blob}}'s {{Blob/type}} attribute,
-if it is not the empty string.
-
-Along with <a href="#TwoHundredOK">200 OK</a> responses,
-user agents use a Content-Length header [[RFC7230]]
-that is equal to the value of the {{Blob}}'s {{Blob/size}} attribute.
-
-If a Content-Type header [[RFC7231]] is provided,
-then user agents obtain and process that media type
-in a manner consistent with the Mime Sniffing specification [[MIMESNIFF]].
-
-If a resource identified with a <a>Blob URL</a> is a {{File}} object,
-user agents use that file's {{File/name}} attribute,
-as if the [=/response=] had a <code>Content-Disposition</code> header
-with the filename parameter set to the {{File}}'s {{File/name}} attribute [[!RFC6266]].
-
-Note: A corollary to this is a non-normative implementation guideline:
-that the "File Save As" user interface in user agents takes into account the {{File}}'s {{File/name}} attribute
-if presenting a default name to save the file as.
-
-<h4 id="NetworkError">
-Network Errors</h4>
-
-Responses that do not succeed with a <a href="#TwoHundredOK">200 OK</a>
-act as if a <a>network error</a> has occurred.
-Network errors are used when:
+Issue: This more or less matches chrome's behavior,
+but other browsers seem to not only ignore fragments,
+but also ignore |url|'s [=url/query=],
+and possibly even extra [=url/path=] components.
+</div>
 
-* Any [=/request=] method other than GET is used.
-* The <a>Blob URL</a> does not have an entry in the <a>Blob URL Store</a>.
-* A [=CORS protocol|cross-origin requests=] is made on a <a>Blob URL</a>.
-* A security error has occurred.
+Futher requirements for the parsing an fetching model for [=Blob URLs=] are defined in the [[!URL]] and [[!Fetch]] specifications.
 
-<h4 id="ProtocolExamples">
-Sample Request and Response Headers</h4>
+<h4 id="originOfBlobURL">
+Origin of Blob URLs</h4>
 
-*This section is informative.*
+<div algorithm="resolveURLOrigin">
+To <dfn for="Blob URL" lt="resolve the origin">resolve the origin of a Blob URL</dfn> given a |url| (a [=URL=]), run the following steps:
 
-This section provides sample exchanges between web applications and user agents using <a>Blob URLs</a>.
-A [=/request=] can be triggered using HTML markup of the sort
-<code class="lang-markup">&lt;img src="blob:http://example.org:8080/550e8400-e29b-41d4-a716-446655440000"></code>.
-These examples merely illustrate the [=/request=] and [=/response=];
-web developers are not likely to interact with all the headers,
-but the {{XMLHttpRequest/getAllResponseHeaders()}} method of {{XMLHttpRequest}}, if used,
-will show relevant [=/response=] headers.
+1. [=Assert=]: |url|'s [=url/scheme=] is <code>"blob"</code>.
+1. Let |entry| be the result of [=Blob URL/resolve|resolving=] |url|.
+1. If |entry| is not undefined, return |entry|'s [=Blob URL Entry/environment=]'s [=environment settings object/origin=].
```

Comment:
Hmm, so unlike JS, we don't say that getting something from a map returns undefined when it doesn't exist. Instead you should probably do something like "If map[x] exists, return map[x]; otherwise, return failure." Then test for failure here. Or null instead of failure, perhaps.

Reply:
Hmm, I guess I misremembered Infra. Well, yeah, this definitely needs clarity; "nothing" is pretty ambiguous. I will open a new issue.